### PR TITLE
Feature/60

### PR DESCRIPTION
**Pull request by Shirashoji**
This pull request introduces several enhancements and updates to the project, including dependency updates, state management improvements using Recoil, and UI enhancements for concert management. The most important changes include updating dependencies in `package.json`, implementing Recoil state management, adding a modal for concert management, and updating the `MyConcertCard` component to handle new state and UI logic.

### Dependency Updates:
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L13-R17): Updated dependencies for `@emotion/react`, `@emotion/styled`, `@fontsource/roboto`, `@mui/icons-material`, and `@mui/material` to their latest versions.

### State Management with Recoil:
* [`src/components/RecoilStates.jsx`](diffhunk://#diff-a5c21e8f8c716637a904b04b78d9f3a2e258f8053b34cac08e4292ab17cd6013R1-R71): Added Recoil atoms and selectors for managing concert states, including `concertsState`, `workConcertState`, `concertNamesState`, `selectedConcertState`, and `concertListState`.

### UI Enhancements:
* [`src/components/layouts/AddMyConcert.jsx`](diffhunk://#diff-1b5d0a9ef044a568460cdafa71ab47ca10b41ed0f1d822dd58905d05d3e705ebR1-R184): Added a new modal component for managing concert details, including selecting movements and handling state updates.
* [`src/components/layouts/MyConcert/Card.jsx`](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL23-L27): Updated the `MyConcertCard` component to integrate with the new Recoil states, handle movement selection, and manage the concert details modal. [[1]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL23-L27) [[2]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL69-R91) [[3]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL87-R102) [[4]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL146-R161) [[5]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL167-R182) [[6]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL210-R229) [[7]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL222-R245) [[8]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaR260-R277) [[9]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL261-R298) [[10]](diffhunk://#diff-15e7a3c77d484ab4a68fa48bdf611fb5ea7e3806ffe96158609157701954acdaL285-R322)

### Minor Fixes:
* [`src/components/layouts/FilterDialog.jsx`](diffhunk://#diff-94f6617209c5da84801423f30b21d478b6d7df48c3e647268a76b9d6ab008767L44-R44): Fixed a syntax issue in the `applyFilter` function.

### Additional Updates:
* [`src/components/layouts/MyConcert/ConcertMenus.jsx`](diffhunk://#diff-61681833d27e07ea2a8b0fee5cd59df20f5134aee1108ef0185e1ac2cfb7db15R7-R19): Added functionality for sharing concert details via URL parameters and copying to the clipboard. [[1]](diffhunk://#diff-61681833d27e07ea2a8b0fee5cd59df20f5134aee1108ef0185e1ac2cfb7db15R7-R19) [[2]](diffhunk://#diff-61681833d27e07ea2a8b0fee5cd59df20f5134aee1108ef0185e1ac2cfb7db15L53-R67) [[3]](diffhunk://#diff-61681833d27e07ea2a8b0fee5cd59df20f5134aee1108ef0185e1ac2cfb7db15L88-R95) [[4]](diffhunk://#diff-61681833d27e07ea2a8b0fee5cd59df20f5134aee1108ef0185e1ac2cfb7db15L100-R111) [[5]](diffhunk://#diff-61681833d27e07ea2a8b0fee5cd59df20f5134aee1108ef0185e1ac2cfb7db15R143-R178)

### CODE REVIEW

File: src/components/layouts/SplitButton.jsx [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
-  const mainConcertID = useRecoilValue(selectedConcertState);
-  const setConcerts = useSetRecoilState(workConcertState);
 
+  const setWorkConcerts = useSetRecoilState(workConcertState);
+  const [openModal, setOpenModal] = React.useState(false);
+
+  const [concertID, setConcertID] = React.useState("");
+
+  const mainConcertID = useRecoilValue(selectedConcertState);
   const mainConcert = concerts.find((concert) => concert.id === mainConcertID);
```
Comment (aramackie) [inline src/components/layouts/SplitButton.jsx:31]: - L.30 で mainConcertID = undefined のままこの行に入ると、mainConcert も undefined になっちゃう

Comment (aramackie) [inline src/components/layouts/SplitButton.jsx:31]: L.30 で mainConcertID = undefined のままこの行に入ると、mainConcert も undefined になっちゃう

**Review by aramackie — CHANGES_REQUESTED** [review]
L.30でmainConcertID が存在しない値の場合（初回アクセス時など）は、L.31で mainConcert = undefined となるので、L.102で mainConcert.id が読み取れずにクラッシュしちゃう🥺

**Comment by Shirashoji** [comment]
@aramackie

**Pull request merged**